app/src/screens/wallet/WalletPaymentMethodScreen.tsx

```diff
@@ -0,0 +1,93 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import React, { useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+// @ts-ignore
+import { CreditCardInput } from 'react-native-credit-card-input';
+
+import { Container } from '../../components/layout/Container';
+import { PaddedInputScrollView } from '../../components/layout/PaddedInputScrollView';
+import { AppButton } from '../../components/shared/AppButton';
+import { useUpdateUserPaymentMethodMutation } from '../../hooks/payments/useUpdateUserPaymentMethodMutation';
+import { Stripe } from '../../lib/stripe';
+import { CreditCardInputResult } from '../../types/credit-card-input-result';
+import { WalletScreenStackParamList } from './WalletScreenStack';
+
+type WalletPaymentMethodScreenProps = StackScreenProps<
+  WalletScreenStackParamList,
+  'PaymentMethod'
+>;
+
+const preparePaymentMethodPayload = (
+  card: CreditCardInputResult
+): CreatePaymentMethodPayload => {
+  const { number, expiry, cvc } = card.values;
+  const [exp_month, exp_year] = expiry.split('/');
+
+  return {
+    type: 'card',
+    card: {
+      number,
+      cvc,
+      exp_month,
+      exp_year,
+    },
+  };
+};
+
+export const WalletPaymentMethodScreen: React.FC<WalletPaymentMethodScreenProps> = ({
+  navigation,
+}) => {
+  const [card, setCard] = useState<CreditCardInputResult | null>(null);
+  const [updatePaymentMethod] = useUpdateUserPaymentMethodMutation();
+  const [isPending, setPending] = useState(false);
+
+  const onSave = async () => {
+    if (!card || !card.valid) {
+      alert('Wprowadź prawidłowe dane');
+      return;
+    }
+
+    setPending(true);
+
+    try {
+      const response = await Stripe.createPaymentMethod(
+        preparePaymentMethodPayload(card)
+      );
+      const { id } = response.data;
+      await updatePaymentMethod(id, { throwOnError: true });
+      navigation.navigate('Wallet');
+    } catch (error) {
+      alert(
+        'Podczas zapisywania metody płatności wystąpił błąd. Spróbuj ponownie lub skontaktuj się z supportem.'
+      );
+      console.log(error.response);
+    } finally {
+      setPending(false);
+    }
+  };
+
+  return (
+    <Container>
+      <PaddedInputScrollView>
```

Comment:
Didn't we swap out this library? I guess the name of the component stayed the same :)